experiments/pam/pam_cortx.c

```diff
@@ -29,118 +32,182 @@
 // libjson
 #include <json-c/json.h>
 
+#ifndef PAM_EXTERN
+#define PAM_EXTERN
+#endif
+
 /* expected hook */
-PAM_EXTERN int pam_sm_setcred(pam_handle_t *pamh, int flags, int argc, const char **argv)
-{
-	pam_syslog(pamh, LOG_INFO, "Initialized SM SetCredentials\n");
+PAM_EXTERN int pam_sm_setcred(pam_handle_t *pamh, int flags, int argc, const char **argv){
 	return PAM_SUCCESS;
 }
 
-PAM_EXTERN int pam_sm_acct_mgmt(pam_handle_t *pamh, int flags, int argc, const char **argv)
-{
-	pam_syslog(pamh, LOG_INFO, "Initialized SM Acc Management\n");
-	return PAM_SUCCESS;
+PAM_EXTERN int pam_sm_acct_mgmt(pam_handle_t *pamh, int flags, int argc, const char **argv){
+	pam_syslog(pamh, LOG_ERR, "Acc management is not supported\n");
+	return PAM_SERVICE_ERR;
 }
+
 PAM_EXTERN int pam_sm_chauthtok(pam_handle_t *pamh, int flags, int argc, const char **argv){
-pam_syslog(pamh, LOG_INFO, "Initialized CHAAUTOK\n");
-	return PAM_SUCCESS;
+	pam_syslog(pamh, LOG_ERR, "Password management is not supported\n");
+	return PAM_SERVICE_ERR;
+}
+
+PAM_EXTERN int pam_sm_open_session(pam_handle_t *pamh, int flags, int argc, const char **argv){
+	pam_syslog(pamh, LOG_ERR, "Session management is not supported\n");
+	return PAM_SERVICE_ERR;
 }
-static int login(pam_handle_t *pamh, const char *pUsername, const char *pPassword)
-{
-    //Login Function for CSM Initialization.
-	pam_syslog(pamh, LOG_INFO, "Verify CSM User %s \n", pUsername);
-	static const char *pUrl = "http://localhost:28101/api/v1/login";
-	CURL *pCurl = curl_easy_init();
+
+PAM_EXTERN int pam_sm_close_session(pam_handle_t *pamh, int flags, int argc, const char **argv){
+	pam_syslog(pamh, LOG_ERR, "Session management is not supported\n");
+	return PAM_SERVICE_ERR;
+}
+
+static void curl_cleanup(struct curl_slist *headers, CURL *pCurl){
+	curl_slist_free_all(headers);
+	curl_easy_cleanup(pCurl);
+}
+
+static int login(pam_handle_t *pamh, const char *pUsername, const char *pPassword,
+				uid_t uid, gid_t gid){
+    //Login Function for CSM/S3 users
+	const char *pUrl = "http://localhost:28101/api/v2/login";
+	CURL *pCurl = NULL;
 	struct curl_slist *headers = NULL;
 	int res = -1;
-	static const char *header_file_name = "/var/log/response_headers.out";
+	char body_file_name[100] = "/tmp/response_body_";
```

Comment:
Why 100 bytes? Why cant char body_file_name[] = "/tmp/response_body_";

Reply:
@pawankumarsrivastava 
I checked without allocating fixed number of bytes. But was getting *** stack smashing detected ***. And file names were not getting created proper. eg. /tmp/response_body_**s3#&02**. 
That is why allocated fixed bytes considering the max number of bytes required for csm/s3 username.